net_transport.go

```diff
@@ -721,14 +760,25 @@ func sendRPC(conn *netConn, rpcType uint8, args interface{}) error {
 	return nil
 }
 
-// newNetPipeline is used to construct a netPipeline from a given
-// transport and connection.
-func newNetPipeline(trans *NetworkTransport, conn *netConn) *netPipeline {
+// newNetPipeline is used to construct a netPipeline from a given transport and
+// connection. It is a bug to ever call this with maxInFlight less than 2 and
+// will cause a panic.
+func newNetPipeline(trans *NetworkTransport, conn *netConn, maxInFlight int) *netPipeline {
+	if maxInFlight < 2 {
+		// Shouldn't happen (tm) since we validate this in the one call site and
+		// skip pipelining if it's lower.
+		panic("pipelining makes no sense if maxInFlight < 2")
```

Comment:
Not blocking: I'm ok with this implementation, but how do we feel about returning an error here instead? Throwing a panic in a library doesn't give the caller many options, and I don't _think_ this is fatal?

Reply:
We could... but if this is ever hit it implies a buggy change to this library violating the current invariants of the valid values here.

Returning an error is possible but would involve plumbing an error back through the return of this function and back through the Transport interface somehow which seemed like a bunch of lifting to handle a case that can't ever happen. I'm open to it though i you or anyone feels like it's safer?